### DISCUSSION TITLE
Could not find module

### DISCUSSION THREAD

**Discussion by raulaguila** [marcelotduarte/cx_Freeze#1348] [Q&A] [answered]
Hello,

I am creating an application with <a href="https://pypi.org/project/pylibdmtx/">pylibdmtx</a> library. 

This application is working correctly in my computer when i create and install a windows installer with cx_Freeze, however, when I send this application to a new computer (a virtual machine for example), I can install it normally, but when I run it, it gives this error:

<img src="https://github.com/raulaguila/APIService/blob/main/assets/img/erro.png?raw=true" alt="Module Error">

(This files directory "C:\Users\raul4\Desktop" is in my computer, not in the virtal machine. is this really correct?)
I already checked and the file that says it doesn't exist, yes exist, as i showing in this print:

<img src="https://github.com/raulaguila/APIService/blob/main/assets/img/directory.png?raw=true" alt="File Error">

My 'build.py' code:

```
packages_include = [
    "os",
    "logging.config",
    "pylibdmtx"
]

modules_exclude = [
    "Tkinter"
]

files_include = [
    os.path.join(os.getcwd(), 'assets'),
    os.path.join(os.getcwd(), 'configs')
]

if sys.platform == "win32":

    app_icon = app_icon.replace('\\', '\\\\')
    app_main = app_main.replace('\\', '\\\\')
    app_platform = "Win32GUI"

modules_include: list = []

for folder in os.listdir(os.path.join(os.getcwd(), 'src')):
    if folder != '__pycache__' and os.path.isdir(os.path.join(os.path.join(os.getcwd(), 'src'), folder)):
        modules_include.append(f'src.{folder}')
        for module in os.listdir(os.path.join(os.path.join(os.getcwd(), 'src'), folder)):
            if module != '__pycache__' and '.py' in module:
                modules_include.append(f'src.{folder}.{module[:-3]}')

cx_Freeze.setup(
    name=app_name,
    author=app_author,
    author_email=app_email,
    version=app_version,
    description=app_description,
    options={"build_exe": {
        "packages": packages_include,
        "excludes": modules_exclude,
        "includes": modules_include,
        "include_files": files_include
    }},
    executables=[cx_Freeze.Executable(
        script=app_main,
        base=app_platform,
        initScript=None,
        target_name=app_name_target,
        shortcut_name=app_name_shortcut,
        shortcut_dir="DesktopFolder",
        icon=app_icon
    )]
)
```

I found many "solutions" on the internet but none solved my problem, can someone help me please?

**Comment by marcelotduarte** [comment, marked as answer]
> (This files directory "C:\Users\raul4\Desktop" is in my computer, not in the virtal machine. is this really correct?)

Yes, it is. To change this, see [replace_paths](https://cx-freeze.readthedocs.io/en/latest/setup_script.html#cmdoption-arg-replace_paths)

> Could not find file (or one of its dependencies)

libmtdx-64.ddl is compiled in MSVC, so probably it depend of its runtime, so try [include_msvcr](https://cx-freeze.readthedocs.io/en/latest/setup_script.html#cmdoption-arg-include_msvcr)
More info: [FAQ](https://cx-freeze.readthedocs.io/en/latest/faq.html#microsoft-visual-c-redistributable-package)

**Reply by raulaguila** [reply]
Problem solved.
I didn't pay attention and I didn't install <a href="https://www.microsoft.com/en-US/download/details.aspx?id=40784">Visual C++ Redistributable Packages</a>.
You are a god!
Thanks!

**Reply by raulaguila** [reply]
@marcelotduarte Is possible change de MSVC version in cx_Freeze?

**Reply by marcelotduarte** [reply]
I don't know if I understand your question. You can either generate the executable without msvcr and install the runtime manually or I have already included the runtime using the include_msvcr option if the conditions explained in the FAQ are met. If that's not what you want to know, feel free to ask.
